s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java

```diff
@@ -63,7 +66,16 @@ public void process() throws Exception {
                 };
                 executor.execute(runnable);
             } catch (Exception ex) {
-                logger.log(Level.SEVERE, "Failed to initialize S2Cache options."+ex.getMessage());
+                logger.log(Level.SEVERE, "Failed to initialize S2Cache options.",ex);
             }
     }
+
+    public void complete() {
+        try {
+            executor.complete();
```

Comment:
I don't think this does anything. `ThreadExecutor.complete()` calls `ThreadPoolExecutor.shutdown()`, which not only doesn't interrupt the `Thread.sleep` call, but also doesn't kill the running tasks:

> Initiates an orderly shutdown in which previously submitted tasks are executed, but no new tasks will be accepted.

Generally, killing running threads is a finicky operations, hence my suggestion to use a timer.

Reply:
Yes, one one my reasons for the PR